cmd/internal/cli/build.go

```diff
@@ -40,8 +40,37 @@ var (
 	dockerPassword string
 	dockerLogin    bool
 	noCleanUp      bool
+
+	// AllowUnauthenticatedBuild When true, will never ask if you want to build
+	// a container from a unsigned base container. This only works when your bootstrap
+	// is the library
+	AllowUnauthenticatedBuild bool
+	// LocalVerifyBuild if true, ...
+	LocalVerifyBuild bool
 )
 
+// --allow-unsigned, -U
+var buildAllowUnsignedFlag = cmdline.Flag{
+	ID:           "buildAllowUnsignedFlag",
+	Value:        &AllowUnauthenticatedBuild,
+	DefaultValue: false,
+	Name:         "allow-unsigned",
+	ShortHand:    "U",
+	Usage:        "never prompt when building from a unsigned container, only when bootstrap is library",
+	EnvKeys:      []string{"ALLOW_UNSIGNED"},
```

Comment:
there's something interesting going on here: this environment variable is already defined for push and pull. The name of the corresponding flag is the name. To me it seems it's the same flag in all cases, but they all associate different Go variables with the flags. The end effect in the UI is that you can set a single environment variable `SINGULARITY_ALLOW_UNSIGNED=true` and this will set all three flags.

This is either intentional, in which case I would suggest adding a comment to document that it is intentional, or an accident, in which case I would suggest changing the name of the environment variable to something like SINGULARITY_BUILD_ALLOW_UNSIGNED.